datumaro/plugins/ade20k2020_format.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (C) 2020-2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from itertools import chain
+import glob
+import json
+import logging as log
+import os
+import os.path as osp
+
+import numpy as np
+
+from datumaro.components.converter import Converter
+from datumaro.components.extractor import (
+    AnnotationType, CompiledMask, DatasetItem, Extractor, Importer,
+    LabelCategories, Mask, Polygon,
+)
+from datumaro.util.image import find_images, lazy_image, load_image
+
+
+class Ade20k2020Extractor(Extractor):
+    def __init__(self, path):
+        if not osp.isdir(path):
+            raise FileNotFoundError("Can't read dataset directory '%s'" % path)
+
+        subsets = os.listdir(path)
+        if len(subsets) < 1:
+            raise FileNotFoundError("Can't read subsets in directory '%s'" % path)
+
+        super().__init__(subsets=sorted(subsets))
+        self._path = path
+
+        self._items = []
+        self._categories  = {}
+
+        for subset in self._subsets:
+            self._load_items(subset)
+
+    def __iter__(self):
+        return iter(self._items)
+
+    def categories(self):
+        return self._categories
+
+    def _load_items(self, subset):
+        labels = self._categories.setdefault(AnnotationType.label,
+            LabelCategories())
+        path = osp.join(self._path, subset)
+
+        images = [i for i in find_images(path, '.jpg', recursive=True)]
+
+        for image_path in sorted(images):
+            item_id = osp.splitext(osp.basename(image_path))[0]
```

Comment:
Using only basenames can lead to name conflict, if images have different directory names.